src/test/java/com/cronutils/model/time/ExecutionTimeQuartzWithDayOfYearExtensionIntegrationTest.java

```diff
@@ -43,47 +42,52 @@ public void setUp() throws Exception {
+
+        for (int i = 1; i < 30; i++) {
+            final ZonedDateTime now = truncateToDays(ZonedDateTime.of(2017, 10, i, 0, 0, 0, 0, UTC));
+            final int dayOfMostRecentPeriod = (now.getDayOfYear() - 1) % 14;
+            final ZonedDateTime expected = now.plusDays(14 - dayOfMostRecentPeriod);
+            assertEquals("Wrong next time from " + now, expected, executionTime.nextExecution(now).get());
+        }
     }
 
     @Test
     public void testNextExecutionEveryTwoWeeksStartingWithFirstDayOfYearIssue249() {
-        ZonedDateTime now = truncateToDays(ZonedDateTime.now().minusDays(1));
-        int dayOfYear = now.getDayOfYear();
-        int dayOfMostRecentPeriod = dayOfYear % 14;
-        ZonedDateTime expected = now.plusDays(15 - dayOfMostRecentPeriod);
         ExecutionTime executionTime = ExecutionTime.forCron(parser.parse(BI_WEEKLY_STARTING_WITH_FIRST_DAY_OF_YEAR));
-        assertEquals(expected, executionTime.nextExecution(now).get());
+
+        for (int i = 1; i < 30; i++) {
+            ZonedDateTime now = truncateToDays(ZonedDateTime.of(2017, 10, i, 0, 0, 0, 0, UTC));
+            int dayOfMostRecentPeriod = (now.getDayOfYear() - 1) % 14;
+            ZonedDateTime expected = now.plusDays(14 - dayOfMostRecentPeriod);
+            assertEquals("Wrong next time from " + now, expected, executionTime.nextExecution(now).get());
+        }
     }
 
     @Test
     public void testLastExecutionEveryTwoWeeksStartingWithFirstDayOfYear() {
-        final ZonedDateTime now = truncateToDays(ZonedDateTime.now().minusDays(1));
-        final int dayOfYear = now.getDayOfYear();
-        final int dayOfMostRecentPeriod = dayOfYear % 14;
-        final ZonedDateTime expected = dayOfMostRecentPeriod == 1 ? now.minusDays(14) : now.minusDays(dayOfMostRecentPeriod - 1);
         final ExecutionTime executionTime = ExecutionTime.forCron(parser.parse(BI_WEEKLY_STARTING_WITH_FIRST_DAY_OF_YEAR));
-        assertEquals(expected, executionTime.lastExecution(now).get());
+
+        for (int i = 1; i < 30; i++) {
+            ZonedDateTime now = truncateToDays(ZonedDateTime.of(2017, 10, i, 0, 0, 0, 0, UTC));
+            final int dayOfMostRecentPeriod = (now.getDayOfYear() - 1) % 14;
+            final ZonedDateTime expected = now.minusDays(dayOfMostRecentPeriod == 0 ? 14 : dayOfMostRecentPeriod);
+            assertEquals("Wrong next time from " + now, expected, executionTime.lastExecution(now).get());
+        }
     }
 
     @Test
     public void testLastExecutionEveryTwoWeeksStartingWithFirstDayOfYearIssue249() {
         //s m H DoM M DoW Y DoY
-        ZonedDateTime now = truncateToDays(ZonedDateTime.of(2017, 10, 7, 0, 0, 0, 0, ZoneId.of("America/Argentina/Buenos_Aires")));
-        int dayOfYear = now.getDayOfYear();
-        ZonedDateTime expected = truncateToDays(ZonedDateTime.of(2017, 9, 24, 0, 0, 0, 0, ZoneId.of("America/Argentina/Buenos_Aires")));
+        ZonedDateTime now = ZonedDateTime.of(2017, 10, 7, 0, 0, 0, 0, UTC);
```

Comment:
Is this change of TimZone really fixing this test? This seems weird.

Reply:
The time zone change is not fixing the test cases, but it does align them to the tests around them.
I felt it a bit arbitrary to have UTC in some places and Buenos_Aires in other.

Reply:
@SvenssonWeb @Naxos84 is not an issue to have different time zones when setting up a scenario. The issue is if we don't have a timezone at all, which makes test setup and results depend on the place of execution. 
If we see that we have a test failing with a given timezone and not with another one, then we have a bug, and should rather add a test than leave one out 😄 
Thank you for digging into this!